draft-ietf-pquip-pqc-engineers.md

```diff
@@ -325,29 +356,30 @@ It is noteworthy that lattice-based encryption schemes require a rounding step d
 
 ## Hash-Based Public-Key Cryptography {#hash-based}
 
-Hash based PKC has been around since the 1970s, when it was developed by Lamport and Merkle. It is used to create digital signature algorithms and its security is mathematically based on the security of the selected cryptographic hash function. Many variants of hash-based signatures (HBS) have been developed since the 70s including the recent XMSS {{!RFC8391}}, HSS/LMS {{!RFC8554}} or BPQS schemes. Unlike digital signature techniques, most hash-based signature schemes are stateful, which means that signing necessitates the update and careful tracking of the secret key. Producing multiple signatures using the same secret key state results in loss of security and ultimately signature forgery attacks against that key.
+Hash based PKC has been around since the 1970s, when it was developed by Lamport and Merkle. It is used to create digital signature algorithms and its security is mathematically based on the security of the selected cryptographic hash function. Many variants of hash-based signatures (HBS) have been developed since the 70s including the recent XMSS {{!RFC8391}}, HSS/LMS {{!RFC8554}} or BPQS schemes. Unlike digital signature techniques, most hash-based signature schemes are stateful, which means that signing necessitates the update and careful tracking of the secret key. Producing multiple signatures using the same secret key state results in loss of security and ultimately signature forgery attacks against that key. This leads to complications and operational complexity where stateful hash-based keys need to have long operational lifetimes. For example consider a 20-year root key; there is an expectation that 20 years is longer than the expected lifetime of the hardware that key is stored on, and therefore the key will need to be migrated to new hardware at some point. This directly conflicts with requirements to never duplicate the private key in order to prevent the possibility of state re-use. At the time of writing, some proprietary solutions exist for solving this problem, but it is generally accepted that long-lived stateful HBS keys require a large amount of operational consideration before deployment.
```

Comment:
"directly conflict" is too strong, given that there are fairly straightforward solutions that have been in the field for years now.  It is true that a large amount of operational consideration is necessary, but there's no direct conflict with any requirements, other than "never duplicate the private key", which can't be a requirement for exactly the reasons you point out.

I think it should be reworded just a bit to point out that naive implementations which ignore operational issues are disasters waiting to happen, which I think is the point you were trying to make.

Reply:
The secure firmware update mechanism discussed in the SUIT WG leverages HSS. If the device's life expectancy is shorter than the key's expiry, why would the key need to be migrated to a new device ?

Reply:
@timfromdigicert  point taken. I believe that the straightforward solutions that have been in the field for years now are in violation of SP.800-208. That's the "direct conflict" that I had in mind. But we expect 800-208 to change in the next few years, so I'll soften this wording.

@tireddy2 
> If the device's life expectancy is shorter than the key's expiry, why would the key need to be migrated to a new device ?

It wouldn't. I'll make this more clear.